checks/__init__.py

```diff
@@ -390,6 +391,12 @@ def set_check_version(self, manifest=None):
 
         self.check_version = version
 
+    def get_instance_proxy(self, instance, uri):
+        proxies = self.proxies.copy()
+        proxies['no'] = get_no_proxy_from_env()
+
+        return config_proxy_skip(proxies, uri, _is_affirmative(instance.get('skip_proxy', False)))
```

Comment:
We could add the code in `utils` here, but since they're functions that operate over the proxy dicts we pass on to `requests` they have some utility out of the `AgentCheck()` scope.